melodies_monet/util/metcalc.py

```diff
@@ -0,0 +1,789 @@
+    # print("wspd_np shape:", wspd_np.shape)
+
+    # fix is needed in order to work with vert plots. Check dimensions before proceeding. 
+    if hasattr(obj, "coords") and hasattr(obj, "dims"): 
+        dims_to_use = list(u.dims)
+        if wspd_np.shape != u.shape:
+            try:
+                axes_needed = [wspd_np.shape.index(s) for s in u.shape]
+                wspd_np = np.transpose(wspd_np, axes_needed)
+                #print("Transposed wspd_np shape:", wspd_np.shape)
+            except Exception as e:
+                print("Error transposing wspd_np:", e)
+                raise ValueError(
+                    f"Could not align output shape {wspd_np.shape} to input shape {u.shape}; "
+                    "please check dimension alignment."
+                )
+        obj[output_key] = (dims_to_use, wspd_np)
+        obj[output_key].attrs["units"] = "m/s"
+        return obj
+    elif isinstance(obj, dict):
+        obj[output_key] = wspd_np
+        return obj
+    else:
+        return wspd_np
+
+# calc wind direction
+def wdir(obj, varmap = None, output_key = "winddir"):
+    # grab variable names from the yaml 
+    u_key = varmap["u_comp"] 
+    v_key = varmap["v_comp"] 
+    
+    u = obj[u_key]
+    v = obj[v_key]
+
+    # metpy version of this is throwing in weird dimensions so calc by hand
+    # wdr_rad = np.arctan2(-v, -u)
+    wdr_rad = np.arctan2(u, v)
+    #wdr_deg = np.degrees(wdr_rad)
+    wdr_deg = (np.degrees(wdr_rad) + 180)
+    winddir = wdr_deg % 360
+    #print(winddir)
+    winddir_np = winddir.astype("float64").values
+
+    if hasattr(obj, "coords") and hasattr(obj, "dims"):  
+        dims_to_use = list(u.dims)
+        if winddir_np.shape != u.shape:
+            try:
+                axes_needed = [winddir_np.shape.index(s) for s in u.shape]
+                winddir_np = np.transpose(winddir_np, axes_needed)
+                #print("Transposed winddir_np shape:", winddir_np.shape)
+            except Exception as e:
+                print("Error transposing winddir_np:", e)
+                raise ValueError(
+                    f"Could not align output shape {winddir_np.shape} to input shape {u.shape}; "
+                    "please check dimension alignment."
+                )
+        obj[output_key] = (dims_to_use, winddir_np)
+        obj[output_key].attrs["units"] = "degrees"
+        return obj
+    elif isinstance(obj, dict):
+        obj[output_key] = winddir_np
+        return obj
+    else:
+        return winddir_np
+
+# calc potential temperature
+def ptemp(obj, varmap=None, output_key="ptemp", default_keys=None):
+    if default_keys is None:
+        default_keys = {
+            "pressure": "pressure",
+            "temperature": "temperature"
+        }
+
+    pressure_key = varmap.get("pres", default_keys["pressure"]) if varmap else default_keys["pressure"]
+    temperature_key = varmap.get("temp", default_keys["temperature"]) if varmap else default_keys["temperature"]
+        
```

Comment:
When I look at your yaml file examples here, you use temp instead of temperature. Your default keys will work of course too. So your code doesn't break, but should "temp" on line 229 actually be "temperature" so that it pulls in the yaml variable name rather than the default?

https://github.com/L12D2/MELODIES-MONET/blob/BoundaryLayerHeight_Tropopause_calc_beta/docs/examples/control_looping_UWY_UFS_CHEM.yaml